Modules/_uuidmodule.c

```diff
@@ -91,45 +121,1730 @@ py_windows_has_stable_node(void)
+        return NULL;
+    }
+    memcpy(self->bytes, bytes, 16);
+    return (PyObject *)self;
+}
+
+static PyObject *
+Uuid_nb_int(PyObject *self)
+{
+    return get_int((uuidobject *)self);
+}
+
+static PyObject *
+Uuid_richcompare(PyObject *self, PyObject *other, int op)
+{
+    uuid_state *state = get_uuid_state_by_cls(Py_TYPE(self));
+
+    if (!PyObject_TypeCheck(other, state->UuidType)) {
+        Py_RETURN_NOTIMPLEMENTED;
+    }
+
+    uuidobject *uuid_self = (uuidobject *)self;
+    uuidobject *uuid_other = (uuidobject *)other;
+
+    int cmp = memcmp(uuid_self->bytes, uuid_other->bytes, 16);
+
+    int result;
+    switch (op) {
+        case Py_EQ:
+            result = (cmp == 0);
+            break;
+        case Py_NE:
+            result = (cmp != 0);
+            break;
+        case Py_LT:
+            result = (cmp < 0);
+            break;
+        case Py_LE:
+            result = (cmp <= 0);
+            break;
+        case Py_GT:
+            result = (cmp > 0);
+            break;
+        case Py_GE:
+            result = (cmp >= 0);
+            break;
+        default:
+            Py_RETURN_NOTIMPLEMENTED;
+    }
+
+    if (result) {
+        Py_RETURN_TRUE;
+    } else {
+        Py_RETURN_FALSE;
+    }
+}
+
+static PyObject *
+Uuid_str(PyObject *self)
+{
+    uuidobject *uuid = (uuidobject *)self;
+
+    // UUID string format: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx (36 chars)
+    char str[36];
+
+    for (int i = 0; i < 4; i++) {
+        byte_to_hex(uuid->bytes[i], &str[i * 2]);
+    }
+    str[8] = '-';
+
+    for (int i = 4; i < 6; i++) {
+        byte_to_hex(uuid->bytes[i], &str[9 + (i - 4) * 2]);
+    }
+    str[13] = '-';
+
+    for (int i = 6; i < 8; i++) {
+        byte_to_hex(uuid->bytes[i], &str[14 + (i - 6) * 2]);
+    }
+    str[18] = '-';
+
+    for (int i = 8; i < 10; i++) {
+        byte_to_hex(uuid->bytes[i], &str[19 + (i - 8) * 2]);
+    }
+    str[23] = '-';
+
+    for (int i = 10; i < 16; i++) {
+        byte_to_hex(uuid->bytes[i], &str[24 + (i - 10) * 2]);
+    }
+
+    return PyUnicode_FromStringAndSize(str, 36);
+}
+
+static PyObject *
+Uuid_repr(PyObject *self)
+{
+    PyObject *str_obj = Uuid_str(self);
+    if (str_obj == NULL) {
+        return NULL;
+    }
+
+    // Get the class name (sadly can't use tp_name -- we don't need the full name)
+    PyObject *cls_name = PyObject_GetAttrString((PyObject *)Py_TYPE(self), "__name__");
+    if (cls_name == NULL) {
+        Py_DECREF(str_obj);
+        return NULL;
+    }
+
+    PyObject *repr = PyUnicode_FromFormat("%U('%U')", cls_name, str_obj);
+    Py_DECREF(str_obj);
+    Py_DECREF(cls_name);
+    return repr;
+}
+
+static int
+Uuid_setattr(PyObject *self, PyObject *name, PyObject *value)
+{
+    PyErr_SetString(PyExc_TypeError, "UUID objects are immutable");
+    return -1;
+}
+
+static PyObject *
+Uuid_get_urn(PyObject *o, void *closure)
+{
+    uuidobject *self = (uuidobject *)o;
+    PyObject *str_obj = Uuid_str((PyObject *)self);
```

Comment:
```suggestion
    PyObject *str_obj = Uuid_str(o);
```

Here you don't need `self`